internal/controller/snapshot/snapshot_adapter.go

```diff
@@ -736,3 +846,87 @@ func (a *Adapter) updateComponentSource(ctx context.Context, c client.Client, co
 	}
 	return nil
 }
+
+func (a *Adapter) prepareGroupSnapshot(application *applicationapiv1alpha1.Application, prGroupHash string) (*applicationapiv1alpha1.Snapshot, []gitops.ComponentSnapshotInfo, error) {
+	applicationComponents, err := a.loader.GetAllApplicationComponents(a.context, a.client, application)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	snapshotComponents := make([]applicationapiv1alpha1.SnapshotComponent, 0)
+	componentSnapshotInfos := make([]gitops.ComponentSnapshotInfo, 0)
+	for _, applicationComponent := range *applicationComponents {
+		applicationComponent := applicationComponent // G601
+		snapshots, err := a.loader.GetMatchingComponentSnapshotsForComponentAndPRGroupHash(a.context, a.client, a.snapshot, applicationComponent.Name, prGroupHash)
+		if err != nil {
+			a.logger.Error(err, "Failed to fetch Snapshots for component", "component.Name", applicationComponent.Name)
+			return nil, nil, err
+		}
+
+		sortedSnapshots := gitops.SortSnapshots(*snapshots)
+		// find the latest component snapshot created for open PR/MR
+		for _, snapshot := range sortedSnapshots {
+			snapshot := snapshot
+			// find the built image for pull/merge request build PLR from the latest opened pull request component snapshot
+			isOpened, err := a.status.IsPRMRInSnapshotOpened(a.context, &snapshot)
+			if err != nil {
+				a.logger.Error(err, "Failed to fetch PR/MR status for component snapshot", "snapshot.Name", a.snapshot.Name)
+				return nil, nil, err
+			}
+			if isOpened {
+				a.logger.Info("PR/MR in snapshot is opend, will find snapshotComponent and add to groupSnapshot")
+				snapshotComponent := gitops.FindMatchingSnapshotComponent(&snapshot, &applicationComponent)
+				componentSnapshotInfos = append(componentSnapshotInfos, gitops.ComponentSnapshotInfo{
+					Component:        applicationComponent.Name,
+					BuildPipelineRun: snapshot.Labels[gitops.BuildPipelineRunNameLabel],
+					Snapshot:         snapshot.Name,
+					Namespace:        a.snapshot.Namespace,
+				})
+				snapshotComponents = append(snapshotComponents, snapshotComponent)
+				break
```

Comment:
Will this `break` only break out of [the inner for-loop](https://github.com/konflux-ci/integration-service/pull/842/files#diff-cbf7e6d58cc12a67b910b5b9108ba68d206e75cc8560ca0ab2b19e2d34415702R868) and continue through the rest of the [outer loop](https://github.com/konflux-ci/integration-service/pull/842/files#diff-cbf7e6d58cc12a67b910b5b9108ba68d206e75cc8560ca0ab2b19e2d34415702R858), which would then also append a component from the GCL?

Reply:
You are right, thanks!